cloudbuild.yaml

```diff
@@ -1,31 +1,53 @@
 steps:
   - name: 'gcr.io/cloud-builders/docker'
-    args:
-      [
-        'build',
-        '-t',
-        'us-central1-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:latest',
-        '-t',
-        'us-central1-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:$COMMIT_SHA',
-        '--build-arg',
-        'UPTIME_CHECKER_GIT_REVISION=$COMMIT_SHA',
-        '--cache-from',
-        'us-central1-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:latest',
-        '.',
-      ]
+    id: 'create-builder'
+    waitFor: ['-']
+    args: ['buildx', 'create', '--driver', 'docker-container', '--name', 'container', '--use']
 
   - name: 'gcr.io/cloud-builders/docker'
+    id: 'build-image'
+    waitFor: ['create-builder']
     entrypoint: 'bash'
     args:
       - '-c'
       - |
-        # Only push "latest" tag when building on "main"
-        [ "$BRANCH_NAME" != "main" ] && exit 0
-        docker push us-central1-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:latest
-
-images: [
-  'us-central1-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:$COMMIT_SHA',
-]
+        set -euxo pipefail
+        if [[ "$BRANCH_NAME" == "main" ]]; then
+            docker buildx build \
+                --push \
+                --platform linux/amd64,linux/arm64 \
+                -t $LOCATION-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:$COMMIT_SHA \
+                -t $LOCATION-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:$SHORT_SHA \
+                -t $LOCATION-docker.pkg.dev/$PROJECT_ID/uptime-checker/image:latest \
```

Comment:
One thing to note: On the public Docker image (on DockerHub and hopefully GHCR), `latest` should be only use on the release branch. For main branch, we use `nightly` instead.